rust/agents/processor/src/push.rs

```diff
@@ -122,11 +123,10 @@ impl Pusher {
                         let message = self
                             .db
                             .message_by_leaf_index(index)?
+                            .map(|message| message.message)
                             .ok_or_else(|| eyre!("Missing message for known proof"))?;
-                        let proven = ProvenMessage {
-                            proof,
-                            message: message.to_vec(),
-                        };
+                        debug_assert_eq!(keccak256(&message), *proof.leaf.as_fixed_bytes());
```

Comment:
is the `debug_assert_eq` meant to stay in?

Reply:
ya that's intentional. Debug asserts are dropped from release builds